File: IntentVeActivityGecisleri/app/src/main/java/com/busrayasar/intentveactivitygecisleri/MainActivity.java
package com.busrayasar.intentveactivitygecisleri;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tanimla();
        tiklama();

    }

    public void tanimla() {
        btn = findViewById(R.id.button);
    }

    public void gecisYap() {
        Intent intent = new Intent(this, Activity2.class);
        startActivity(intent);
    }

    public void tiklama() {
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                gecisYap();
            }
        });
    }
}

File: DifferentActivities2/app/src/main/java/com/busrayasar/differentactivities/MainActivity.kt
package com.busrayasar.differentactivities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    //aktiviteler arası veri yollayalım
    fun changePage(view: View){
        //kullanıcı verisi alalım
        //*******************************************************************************
        val editText = findViewById<EditText>(R.id.editText) // editTexti tanımadığı için bunu ekledim
        //****************************************************************************************
        val userInput = editText.text.toString()
        val intent = Intent(applicationContext, MainActivity2::class.java)
        intent.putExtra("Set value:", userInput)
        startActivity(intent) // bununla sayfaya geçiş çalışır

        //SecondActivity::class.java -> ikinci aktiviteye git
    }
}